File: js/1747.js
const fs = require('fs');

const N = +fs.readFileSync('example.txt').toString().trim();

let k = N;
while (true) {
  if (isPrime(k) && isPellindrom(k)) {
    console.log(k);
    break;
  }
  k++;
}

function isPellindrom(n) {
  const s = n.toString();

  let start = 0;
  let end = s.length - 1;

  while (start < end) {
    if (s[start] !== s[end]) return false;
    start++;
    end--;
  }

  return true;
}

function isPrime(n) {
  if (n === 1) return false;
  for (let i = 2; i * i <= n; i++) {
    if (n % i === 0) return false;
  }
  return true;
}

File: c++/1747.cpp
#include <iostream>
#include <string>

using namespace std;

bool isPrime(int n) {
  if (n == 1) return false;
  for (int i = 2; i * i <= n; i++) {
    if (n % i == 0) return false;
  }
  return true;
}

bool isPellindrom(int n) {
  string s = to_string(n);

  int a = 0;
  int b = s.length() - 1;

  while (a < b) {
    if (s[a] != s[b]) return false;
    a++;
    b--;
  }

  return true;
}

int main() {
  freopen("example.txt", "r", stdin);
  int N;
  cin >> N;

  while (true) {
    if (isPrime(N) && isPellindrom(N)) {
      cout << N;
      return 0;
    }
    N++;
  }

  return 0;
}